### PR TITLE
Adds some derives for the inherent data stuff

### PR DESCRIPTION
**Pull request by bkchr**
This makes working with them easier.

### CODE REVIEW

File: core/inherents/src/lib.rs [hunk 1]
```diff
@@ -50,6 +50,7 @@ pub use runtime_primitives::RuntimeString;
 pub type InherentIdentifier = [u8; 8];
 
 /// Inherent data to include in a block.
+#[derive(Clone)]
 pub struct InherentData {
```
Comment (tomusdrw) [inline core/inherents/src/lib.rs]: Could have `Default` as well.

File: core/inherents/src/lib.rs [hunk 2]
```diff
@@ -153,7 +153,7 @@ impl codec::Decode for InherentData {
 ///
 /// When a fatal error occurres, all other errors are removed and the implementation needs to
 /// abbort checking inherents.
-#[derive(Encode, Decode, Clone)]
```
Comment (tomusdrw) [inline core/inherents/src/lib.rs]: I'm actually pretty hesitatnt on this one - Seems that `CheckInherentsResults::new`and `CheckInherentsResults::default` return different values which might be quite misleading.

File: core/inherents/src/lib.rs (hunk 1 above)
Comment (tomusdrw) [reply core/inherents/src/lib.rs]: `::new` could now use `Default::default()` I guess

File: core/inherents/src/lib.rs (hunk 2 above)
Comment (bkchr) [reply core/inherents/src/lib.rs]: Fuu, good catch..

**Pull request merged**